### PR TITLE
API user creation behing a toggle

### PR DESCRIPTION
**Pull request by j0k3r**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Documentation | https://github.com/wallabag/doc/pull/5
| Translation   | yes
| Fixed tickets | https://github.com/wallabag/wallabag/issues/3056
| License       | MIT

Following https://github.com/wallabag/wallabag/pull/3065

I've added a toggle feature (in internal settings) so that user api creation can be disabled while form registration still can be enabled.
Also, the /api/user endpoint shouldn't require authentication. Even if we check the authentication when sending a GET request, to retrieve current user information.

I've moved all the internal settings definition to config to avoid duplicated place to define them.
I don't know why we didn't did that earlier.

@wallabag/translators :globe_with_meridians: can I get the translation for:

> Enable user to be registered using the API

Thanks :+1:

### CODE REVIEW

**Comment by tcitworld** [comment]
> I've moved all the internal settings definition to config to avoid duplicated place to define them.

Maybe a dedicated config file for them ? Otherwise the `config.yml` file is painful to read.

File: src/Wallabag/ApiBundle/Controller/UserRestController.php [hunk 1]
```diff
@@ -43,16 +43,16 @@ public function getUserAction()
      */
     public function putUserAction(Request $request)
     {
-        if (!$this->container->getParameter('fosuser_registration')) {
+        if (!$this->getParameter('fosuser_registration') || !$this->get('craue_config')->get('api_user_registration')) {
```
Comment (tcitworld) [inline src/Wallabag/ApiBundle/Controller/UserRestController.php]:
```php
(!($this->getParameter('fosuser_registration') && $this->get('craue_config')->get('api_user_registration')))
```
?

Comment (j0k3r) [inline src/Wallabag/ApiBundle/Controller/UserRestController.php]: I found it more clear like I did

**Comment by matteocoder** [comment]
@j0k3r  "Abilita la registrazione dell'utente attraverso l'API".

**Comment by tcitworld** [comment]
Since the `wallabag_core` settings were moved to `wallabag.yml` file, we may have to check docs for changes.

**Comment by tcitworld** [comment]
`putUserAction` should return HTTP 201 or HTTP 202.

File: src/Wallabag/CoreBundle/DataFixtures/ORM/LoadSettingData.php [hunk 2]
```diff
@@ -6,173 +6,27 @@
...
-            ],
-            [
-                'name' => 'restricted_access',
-                'value' => '0',
-                'section' => 'entry',
-            ],
-        ];
-
-        foreach ($settings as $setting) {
+        foreach ($this->container->getParameter('wallabag_core.default_internal_settings') as $setting) {
```
Comment (nicosomb) [inline src/Wallabag/CoreBundle/DataFixtures/ORM/LoadSettingData.php]: 👍

**Pull request merged**